### PR TITLE
Fixing nested Transaction bug with Sqlite.

### PR DESCRIPTION
**Pull request by Skrypt**
I have tested with the YesSql.Samples.Hi project.
I've found that now we need to specify in the connection string "Cache=Shared" if we are using a db file.
I also noticed that Sqlite by default should have : 

```C#
cfg.IsolationLevel = IsolationLevel.Serializable
```

P.S. Needs more testing before merging.

### CODE REVIEW

**Comment by sebastienros** [comment]
Have you tried to run all unit tests with the `Demand` call removed ?

**Comment by Skrypt** [comment]
This is what I'm getting when running dotnet test directly from the command shell.
image

File: test/YesSql.Tests/project.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
       "dependencies": {
         "NETStandard.Library": "1.5.0-*",
         "dotnet-test-xunit": "1.0.0-dev-*",
```
Comment (alexbocharov) [inline test/YesSql.Tests/project.json]: @sebastienros @Skrypt It isn't correct 
`"dotnet-test-xunit": "1.0.0-rc2-*"` or `"dotnet-test-xunit": "1.0.0-rc2-173384-19"` 

More information: [aspnetcidev feed](http://myget.org/feed/aspnetcidev/package/nuget/dotnet-test-xunit) or [aspnetcirelease feed](http://myget.org/feed/aspnetcirelease/package/nuget/dotnet-test-xunit)

**Comment by Skrypt** [comment]
I'm getting the same result.

File: test/YesSql.Tests/project.json (hunk 1 above)
Comment (Skrypt) [inline test/YesSql.Tests/project.json]: I'm getting the same result after changing it to either one of those.

Comment (Skrypt) [inline test/YesSql.Tests/project.json]: I think it is related to https://github.com/aspnet/KoreBuild/pull/64

Comment (alexbocharov) [inline test/YesSql.Tests/project.json]: You try to remove `.nuget/packages`. After that `dotnet restore`.

Comment (Skrypt) [inline test/YesSql.Tests/project.json]: I removed these lines in the YesSql.Tests project.json file.

```
    "net451": {
      "frameworkAssemblies": {
        "System.Runtime": {
          "type": "build"
        },
        "System.Reflection": {
          "type": "build"
        },
        "System.Threading.Tasks": {
          "type": "build"
        },
        "System.Runtime.Serialization": "4.0.0.0"
      },
      "dependencies": {
      }
    }
```
And ran tests succesfully : 

image
image


So this is related to : https://github.com/aspnet/KoreBuild/pull/64

**Comment by sebastienros** [comment]
Why did you comment on the commits and not the PR ;) But still, if all the tests pass after that then that's awesome, I'd really like to understand what I intended with a call to `Demand` there.

**Pull request merged**